### PR TITLE
Use workers for tracking operations rather than workerPools

### PR DESCRIPTION
**Pull request by imbstack**
This should avoid us running into issues with the size of entities. It also makes things simpler!

This also lays the groundwork for some other improvements I want to make.

### CODE REVIEW

File: services/worker-manager/src/providers/google.js [hunk 1]
```diff
@@ -40,20 +39,17 @@ class GoogleProvider extends Provider {
...
     this.configSchema = 'config-google';
-    this.fake = fake;
 
     this.instancePermissions = instancePermissions;
     this.project = project;
     this.zonesByRegion = {};
 
-    // TODO: Make fakes be injected
-    if (fake) {
+    if (fakes['google']) {
```
Comment (djmitche) [inline services/worker-manager/src/providers/google.js]: `fakes.google` would work here too..

File: services/worker-manager/src/main.js [hunk 2]
```diff
@@ -181,11 +181,17 @@ let load = loader({
     }),
   },
 
+  // This is used in testing to inject provider fakes
+  externalFakes: {
```
Comment (djmitche) [inline services/worker-manager/src/main.js]: What does "external" mean here?

Comment (imbstack) [reply services/worker-manager/src/main.js]: I was using it to mean "cloud service provider" I guess. Very open to suggestions for better names.

Comment (djmitche) [reply services/worker-manager/src/main.js]: fakeCloudApis?

**Pull request merged**